service/src/main/java/org/sunbird/actor/otp/OTPActor.java

```diff
@@ -240,4 +254,185 @@ private String getType(String type) {
         return null;
     }
   }
+
+
+  /*
+   * This method generates OTP for the provided request.
+   */
+  private void generateOTPV3(Request request) {
+    logger.debug(request.getRequestContext(), "OTPActor:generateOTP method call start.");
+    // Extract necessary parameters from the request.
+    String type = (String) request.getRequest().get(JsonKey.TYPE);
+    String key = (String) request.getRequest().get(JsonKey.KEY);
+    String userId = (String) request.getRequest().get(JsonKey.USER_ID);
+    String contextType = (String) request.getRequest().get(JsonKey.CONTEXT_TYPE);
+    List<String> contextAttributesList = (new ObjectMapper()).convertValue(request.get(JsonKey.CONTEXT_ATTRIBUTES),
+            new TypeReference<>() {
+            });
+    String contextAttributes = String.join(",", contextAttributesList);
+    // If userId is not blank, retrieve email or phone associated with it and update key and type accordingly.
+    if (StringUtils.isNotBlank(userId)) {
+      key = otpService.getEmailPhoneByUserId(userId, type, request.getRequestContext());
+      type = getType(type);
+      // Log the operation.
+      logger.info(
+              request.getRequestContext(),
+              "OTPActor:generateOTP:getEmailPhoneByUserId: called for userId = "
+                      + userId
+                      + " ,key = "
+                      + OTPUtil.maskId(key, type));
+    }
+    // Throttle OTP generation based on rate limit.
+    rateLimitService.throttleByKey(
+            key,
+            type,
+            new RateLimiter[]{OtpRateLimiter.HOUR, OtpRateLimiter.DAY},
+            request.getRequestContext());
+    // Get OTP details based on type, key, contextType, and contextAttributes.
+    String otp;
+    Map<String, Object> details = otpService.getOTPDetailsV3(type, key, request.getRequestContext());
+    // If no details found, generate a new OTP, log its generation, insert the OTP details, and log the insertion.
+    if (MapUtils.isEmpty(details)) {
+      otp = OTPUtil.generateOTP(request.getRequestContext());
+      logger.info(
+              request.getRequestContext(),
+              "OTPActor:generateOTP: new otp generated for Key = "
+                      + OTPUtil.maskId(key, type)
+                      + " & OTP = "
+                      + OTPUtil.maskOTP(otp));
+      otpService.insertOTPDetailsV3(type, key, otp, contextType, contextAttributes, request.getRequestContext());
```

Comment:
Do not send contextAttributes if empty

Reply:
1. If contextAttributes is null then it is treated as empty string and value is saved as empty string.
2. Tested in local is working fine.